usefulUtilities/voteApp/appResources/voteApp.js

```diff
@@ -0,0 +1,1022 @@
+//
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+//
+
+/* globals Tablet, Script, HMD, Controller, AccountServices, Menu */
+
+(function () { // BEGIN LOCAL_SCOPE
+
+    // Modules
+    var AppUi = Script.require('appUi'),
+        GOOGLE_SCRIPTS_URL = Script.require(Script.resolvePath("./resources/secrets/secrets.js")).googleScriptsUrl,
+        URL = Script.resolvePath("./resources/voteApp.html?v12344"),
+        CONFIG = Script.require(Script.resolvePath("./resources/config.js?v12344"));
+
+    // Configurable variables
+    var EVENT_DATE = CONFIG.EVENT_DATE,
+        EVENT_NAME = CONFIG.EVENT_NAME;
+
+    // Unload app variables
+    // Example: UTC month 0 = January ; month 10 = sNovember
+    // year, month, day, hour, minutes
+    var UNLOAD_DATE = CONFIG.UNLOAD_DATE,
+        UNLOAD_TIME = 10000; // show app will unload screen for 10 seconds
+    
+    // App variables
+    var BUTTON_NAME = "VOTE",
+        VOTE_APP_NAME = "voteApp.js";
+
+    // App status variables
+    var firstLoad = true,
+        setupNotLoggedIn = false,
+        FIRST_LOAD_TIMEOUT = 500,
+        REQUEST_READY_STATE = 4;
+
+    // Domains vs Zone visited
+    var DOMAINS_ENABLED = true; // enable domains UI / domain visited checks
+    
+    // Zone visited
+    var intervalCheckVisited,
+        TIME_CHECK = 1000,
+        HAS_VISITED_ZONE_CHECK = false, // if there's a zone to visit
+        GOTO_ADDRESS = "thespot//-26.8224,-11.1442,-67.663/0,0.068694,0,0.997638";
+
+    // Web event types from UI
+    var EVENT_BRIDGE_OPEN_MESSAGE = CONFIG.EVENT_BRIDGE_OPEN_MESSAGE,
+        GOTO_LOCATION = CONFIG.GOTO_LOCATION,
+        GOTO_DOMAIN = CONFIG.GOTO_DOMAIN,
+        VOTE_AVATAR = CONFIG.VOTE_AVATAR,
+        VOTE_DOMAIN = CONFIG.VOTE_DOMAIN;
+
+    // Google scripts type events
+    var VOTE_GOOGLE = "vote",
+        GET_INFO_GOOGLE = "getInfo";
+
+    // Vue UI update event
+    var UPDATE_UI = EVENT_NAME + "_update_ui"; // !important must match Tablet.js
+    
+    // Static strings
+    var DOMAIN = "domain";
+    var AVATAR = "avatar";
+    
+    var DEBUG = true;
+
+    // Local Settings.setValue/getValue key
+    var VOTE_APP_SETTINGS_NAME = EVENT_NAME + "_voteApp"; // "Futvrelands_11_17_2018_voteApp"
+    
+    var DEFAULT_VOTE_APP_SETTINGS = {
+        visited: false,
+        domains: [],
+        voted: {
+            avatar: "", domain: ""
+        }
+    };
+
+    // UI variables
+    var ui;
+    var dataStore = {
+        unload: false,
+        loading: true,
+        loggedin: true,
+
+        visited: false, // set by zone HOLIDAY APP
+
+        voted: {
+            domain: false,
+            avatar: false
+        },
+        openPolls: {
+            avatar: false,
+            domain: false
+        },
+        visitedAllDomains: false, // for domains
+        domains: [],
+        avatars: []
+    };
+    
+
+    var domains = {
```

Comment:
i understand that you are using this to compartmentalize code. i would prefer to see this type of thing compartmentalized into their own modules rather than within variables in one file.

i'm also noticing a lot of similarities between the `domains` variable and the `avatars` domain - it seems like there is a big opportunity here to DRY up a bunch of code

Reply:
**I agree.  If you have any time, I'd like to get your thoughts on how to make it more DRY.** Each one does something similar but have minute changes. Especially with the variables.

Reply:
Let's chat!